simulator/infra/ports/t/topology_test.cpp

```diff
@@ -0,0 +1,146 @@
+/**
+ * Topology unit test
+ * @author Eric Konks
+ */
+
+#include "../module.h"
+#include <catch.hpp>
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/json_parser.hpp>
+
+namespace pt = boost::property_tree;
+
+struct SomeTopology : public BaseTestRoot

```

Comment:
`BaseTestRoot` is not accessible here. I suggest to inherit SomeTopology from Root directly.